packages/expo-facebook/ios/EXFacebook/EXFacebook.m

```diff
@@ -179,21 +182,163 @@ + (NSDictionary *)accessTokenNativeToJSON:(FBSDKAccessToken *)input {
     @"userId": input.userID,
     @"appId": input.appID,
-
+    
     @"permissions": [input.permissions allObjects],
     @"declinedPermissions": [input.declinedPermissions allObjects],
     @"expiredPermissions": [input.expiredPermissions allObjects],
     
     @"expirationDate": @([input.expirationDate timeIntervalSince1970] * 1000),
     @"dataAccessExpirationDate": @([input.dataAccessExpirationDate timeIntervalSince1970] * 1000),
-
+    
     @"refreshDate": @([input.refreshDate timeIntervalSince1970] * 1000),
   };
 }
 
+UM_EXPORT_METHOD_AS(logEventAsync,
+                    logEvent:(NSString *)eventName
+                    valueToSum:(nonnull NSNumber *)valueToSum
+                    parameters:(NSDictionary *)parameters
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  parameters = dictionaryWithNullValuesAsStrings(parameters);
+  
+  [FBSDKAppEvents logEvent:eventName
+                valueToSum:valueToSum
+                parameters:parameters
+               accessToken:nil];
+  resolve(nil);
+}
+
+UM_EXPORT_METHOD_AS(logPurchaseAsync,
+                    logPurchase:(double)purchaseAmount
+                    currency:(NSString *)currency
+                    parameters:(NSDictionary *)parameters
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  parameters = dictionaryWithNullValuesAsStrings(parameters);
+  
+  [FBSDKAppEvents logPurchase:purchaseAmount
+                     currency:currency
+                   parameters:parameters
+                  accessToken:nil];
+  resolve(nil);
+}
+
+UM_EXPORT_METHOD_AS(logPushNotificationOpenAsync,
+                    logPushNotificationOpen:(NSString *)campaign
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSDictionary *payload = @{
+    FBSDKAppEventsPushPayloadKey: @{
+      FBSDKAppEventsPushPayloadCampaignKey: campaign
+    }
+  };
+  [FBSDKAppEvents logPushNotificationOpen:payload];
+  resolve(nil);
+}
+
+UM_EXPORT_METHOD_AS(setUserIDAsync,
+                    setUserID:(NSString *)userID
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  [FBSDKAppEvents setUserID:userID];
+  resolve(nil);
+}
+
+UM_EXPORT_METHOD_AS(getUserIDAsync,
+                    getUserID:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve([FBSDKAppEvents userID]);
+}
+
+UM_EXPORT_METHOD_AS(getAnonymousIDAsync,
+                    getAnonymousID:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  @try {
+    NSString *anonymousID = [FBSDKAppEvents anonymousID];
+    resolve(anonymousID);
+  }
+  @catch (NSError *error) {
+    reject(@"E_ANONYMOUS_ID_ERROR", @"Can not get anonymousID", error);
```

Comment:
```suggestion
    reject(@"ERR_FACEBOOK_ANONYMOUS_ID", @"Can not get anonymousID", error);
```
I'm pretty sure we would like to follow [NodeJS error codes format](https://nodejs.org/api/errors.html#errors_node_js_error_codes) 🤔